### PR TITLE
Event with previous attributes marked invalid

### LINKED ISSUE #489: Webhook message marked invalid because contains previous_attributes
#### Issue Summary

I found that when i make changes to the subscription in the stripe dashboard, the update is not sync'ed by pinax-stripe because the event is marked as invalid in Webhook.py where the received event is compared with the event read from Stripe.

It is kind of odd since we request the same event by it's id but i've seen it for a fact and added a fix + test for it.

---

_If this is a bug instead of a question or feature request, please fill out the sections below._

---

#### Steps to Reproduce

It's essential that you provide enough information for someone else to replicate the problem you're seeing. Simply describing something that's broken on your current project is not enough!

### PR DESCRIPTION
**Pull request by Paul424**
#### What's this PR do?

Event is marked invalid when change originated from Stripe dashboard. The check is now done explicitly on the object part only.

#### What ticket or issue # does this fix?

Closes #489

#### Definition of Done (check if considered and/or addressed):

- [ ] Are all backwards incompatible changes documented in this PR?
- [ ] Have all new dependencies been documented in this PR?
- [ ] Has the appropriate documentation been updated (if applicable)?
- [Y] Have you written tests to prove this change works (if applicable)?

### CODE REVIEW

**Comment by ticosax** [comment]
For reference https://stripe.com/docs/upgrades#2017-05-25
@Paul424 Please can you satisfy the linter ? 
While you are at it, please remove also all unicode prefixes `u""`.
Thank you

**Comment by blueyed** [comment]
@ticosax 
You are referring to the following, aren't you?

> Event objects (and webhooks) now show entire sub-arrays in previous_attributes when those arrays have changes. Previously those sub-arrays only contained the specific fields that were changed.

From that I understand that there was `previous_attributes` before already (which should cause the events to be considered different), no?

**Comment by ticosax** [comment]
I'm not sure about previous versions of stripe. What matters to me is that this PR looks legit according to stripe changelog.

**Comment by blueyed** [comment]
And what I mean is that it is a problem before already.. so even more legit.. ;)

**Comment by Paul424** [comment]
? Hmmm, i just replaced the quotes, now it's failing???

**Comment by Paul424** [comment]
Ok, i reformatted the file, see if that helps...

**Comment by ticosax** [comment]
```diff
--- /home/circleci/repo/pinax/stripe/tests/test_webhooks.py:before	2017-11-09 16:28:19.841512
+++ /home/circleci/repo/pinax/stripe/tests/test_webhooks.py:after	2017-11-09 16:29:08.834824
@@ -34,8 +34,9 @@
     CustomerSubscriptionCreatedWebhook,
     CustomerUpdatedWebhook,
     InvoiceCreatedWebhook,
-    registry,
-    Webhook)
+    Webhook,
+    registry
+)
```
Almost there

**Comment by Paul424** [comment]
Imports are incorrectly sorted; really? What is the correct order?

File: pinax/stripe/tests/test_webhooks.py [hunk 1]
```diff
@@ -44,7 +45,6 @@
 
 
 class WebhookRegistryTest(TestCase):
-
```
Comment (blueyed) [inline pinax/stripe/tests/test_webhooks.py]: You should not change this I guess.
(also below)

File: pinax/stripe/tests/test_webhooks.py [hunk 2]
```diff
@@ -185,7 +185,8 @@ def signal_handler(sender, *args, **kwargs):
     @patch("pinax.stripe.webhooks.Webhook.process_webhook")
     def test_process_exception_is_logged(self, ProcessWebhookMock, ValidateMock, LinkMock):
         # note: we choose an event type for which we do no processing
-        event = Event.objects.create(kind="account.external_account.created", webhook_message={}, valid=True, processed=False)
+        event = Event.objects.create(kind="account.external_account.created", webhook_message={}, valid=True,
+                                     processed=False)
```
Comment (blueyed) [inline pinax/stripe/tests/test_webhooks.py]: Unnecessary change (also below).

File: pinax/stripe/tests/test_webhooks.py [hunk 3]
```diff
@@ -344,23 +345,91 @@ def test_plan_created(self, EventMock):
...
-        event = Event.objects.create(kind=CustomerSubscriptionCreatedWebhook.name, webhook_message={}, valid=True, processed=False)
+        event = Event.objects.create(kind=CustomerSubscriptionCreatedWebhook.name, webhook_message={}, valid=True,
+                                     processed=False)
         CustomerSubscriptionCreatedWebhook(event).process_webhook()
         self.assertFalse(SyncMock.called)
 
 
-class InvoiceCreatedWebhookTest(TestCase):
+class CustomerSubscriptionUpdatedWebhookTest(TestCase):
+    WMD = {"object": {"livemode": False, "id": "my_sub", "billing": "charge_automatically",
```
Comment (blueyed) [inline pinax/stripe/tests/test_webhooks.py]: Please use a more descriptive name, also for `VMD` below.

Comment (Paul424) [reply pinax/stripe/tests/test_webhooks.py]: It's not that i don't agree with you, but jeezzz.... i'm just trying to help here...?

What do you suggest?

Paul

**Comment by paltman** [comment]
@Paul424 thanks!
